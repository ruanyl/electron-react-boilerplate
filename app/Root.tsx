import React from 'react'
import { ConnectedRouter } from 'connected-react-router'
import { History } from 'history'
import { hot } from 'react-hot-loader/root'
import { Provider } from 'react-redux'
import { Store } from 'redux'

import Routes from './Routes'

type Props = {
  store: Store
  history: History
}

const Root = ({ store, history }: Props) => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Routes />
    </ConnectedRouter>
  </Provider>
)

export default hot(Root)
