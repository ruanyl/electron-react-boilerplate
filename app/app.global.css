:global {
  #root {
    width: 100%;
    height: 100%;
  }
}

html {
  width: 100%;
  height: 100%;
}

body {
  background-color: #262626;
  width: 100%;
  height: 100%;
  margin: 0;
}

* {
  box-sizing: border-box;
}
