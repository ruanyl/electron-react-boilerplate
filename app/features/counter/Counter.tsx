import { CounterModel, counterEffects, counterMutations } from './Counter.model'
import { useDispatch, useSelector } from 'react-redux'

import { Link } from 'react-router-dom'
import React from 'react'
import routes from '../../constants/routes.json'
import styles from './Counter.css'

export default function Counter() {
  const dispatch = useDispatch()
  const value = useSelector(CounterModel.selectors.value)
  return (
    <div>
      <div className={styles.backButton} data-tid="backButton">
        <Link to={routes.HOME}>
          <i className="fa fa-arrow-left fa-3x" />
        </Link>
      </div>
      <div className={`counter ${styles.counter}`} data-tid="counter">
        {value}
      </div>
      <div className={styles.btnGroup}>
        <button
          className={styles.btn}
          onClick={() => {
            dispatch(counterMutations.increment())
          }}
          data-tclass="btn"
          type="button"
        >
          <i className="fa fa-plus" />
        </button>
        <button
          className={styles.btn}
          onClick={() => {
            dispatch(counterMutations.decrement())
          }}
          data-tclass="btn"
          type="button"
        >
          <i className="fa fa-minus" />
        </button>
        <button
          className={styles.btn}
          onClick={() => {
            dispatch(counterEffects.incrementIfOdd())
          }}
          data-tclass="btn"
          type="button"
        >
          odd
        </button>
        <button
          className={styles.btn}
          onClick={() => {
            dispatch(counterEffects.incrementAsync())
          }}
          data-tclass="btn"
          type="button"
        >
          async
        </button>
      </div>
    </div>
  )
}
