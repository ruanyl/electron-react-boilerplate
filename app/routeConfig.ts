export const routes = {
  HOME: '/',
  COUNTER: '/counter',
  LOGIN: '/login',
}

export default routes
