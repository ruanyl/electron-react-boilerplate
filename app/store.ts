import modalModule from 'reapex-module-modal'
import immerPlugin from 'reapex-plugin-immer'
import { connectRouter } from 'connected-react-router'
import { routerMiddleware } from 'connected-react-router'
import { createHashHistory } from 'history'
import { App } from 'reapex'
import { createLogger } from 'redux-logger'

const excludeLoggerEnvs = ['test', 'production']
const shouldIncludeLogger = !excludeLoggerEnvs.includes(process.env.NODE_ENV || '')

export const history = createHashHistory()

const router = routerMiddleware(history)
const middlewares = [router]
if (shouldIncludeLogger) {
  const logger = createLogger({
    level: 'info',
    collapsed: true,
  })
  middlewares.push(logger)
}

const routerReducer = connectRouter(history)

export const app = new App({ middlewares })
app.plugin(immerPlugin)

export const modal = app.use(modalModule, '@@modals')

app.registerReducer('router', routerReducer as any)
