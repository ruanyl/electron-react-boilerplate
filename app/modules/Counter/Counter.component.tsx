import React from 'react'
import { Button, Progress } from 'antd'
import { useSelector } from 'react-redux'

import { CounterModel, mutations } from './Counter.model'

export const Counter = () => {
  const total = useSelector(CounterModel.selectors.total)

  return (
    <>
      <Button type="primary" onClick={mutations.decrease}>
        -
      </Button>
      <Progress type="circle" percent={total} />
      <Button type="primary" onClick={mutations.increase}>
        +
      </Button>
    </>
  )
}
