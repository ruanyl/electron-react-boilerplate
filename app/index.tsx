import './app.global.css'

import React, { Fragment } from 'react'
import { app, history } from './store'

import { AppContainer as ReactHotAppContainer } from 'react-hot-loader'
import Root from './containers/Root'
import { render } from 'react-dom'

const store = app.createStore()

const AppContainer = process.env.PLAIN_HMR ? Fragment : ReactHotAppContainer

document.addEventListener('DOMContentLoaded', () =>
  render(
    <AppContainer>
      <Root store={store} history={history} />
    </AppContainer>,
    document.getElementById('root')
  )
)
