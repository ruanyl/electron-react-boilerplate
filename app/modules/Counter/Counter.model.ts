import { app } from '../../store'

export const CounterModel = app.model('Counter', { total: 50 })

export const [mutations] = CounterModel.mutations({
  increase: () => (s) => ({ ...s, total: s.total + 1 }),
  decrease: () => (s) => ({ ...s, total: s.total - 1 }),
})
