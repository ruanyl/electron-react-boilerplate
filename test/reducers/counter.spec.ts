import { CounterModel, counterMutations } from '../../app/features/counter/Counter.model'
import { app } from '../../app/store'

function setup(initialState = { counter: { value: 1 } }) {
  const store = app.createStore(initialState)
  return {
    store,
  }
}

describe('reducers', () => {
  describe('counter', () => {
    it('should handle counterMutations.increment', () => {
      const { store } = setup()
      store.dispatch(counterMutations.increment())
      expect(CounterModel.selectors.self(store.getState())).toMatchSnapshot()
    })

    it('should handle DECREMENT_COUNTER', () => {
      const { store } = setup()
      store.dispatch(counterMutations.decrement())
      expect(CounterModel.selectors.self(store.getState())).toMatchSnapshot()
    })
  })
})
