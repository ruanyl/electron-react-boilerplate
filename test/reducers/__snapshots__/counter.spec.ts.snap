// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle DECREMENT_COUNTER 1`] = `
Object {
  "value": 0,
}
`;

exports[`reducers counter should handle counterMutations.increment 1`] = `
Object {
  "value": 2,
}
`;
